org_fedora_oscap/rule_handling.py

```diff
@@ -668,3 +683,71 @@ def eval_rules(self, ksdata, storage, report_only=False):
             return []
 
     # nothing to be reverted for now
+
+
+class KdumpRules(RuleHandler):
+    """Simple class holding data from the rules affecting the kdump addon."""
+
+    def __init__(self):
+        """Constructor setting the initial value of attributes."""
+
+        self._kdump_enabled = None
+
+    def kdump_enabled(self, kdenabled):
+        """Enable or Disable Kdump"""
+
+        if kdenabled == True:
+            self._kdump_enabled = True
+        if kdenabled == False:
+            self._kdump_enabled = False
+
+    def __str__(self):
+        """Standard method useful for debugging and testing."""
+
+        ret = "kdump"
+
+        if self._kdump_enabled == True:
+            ret += " --enable"
+
+        if self._kdump_enabled == False:
+            ret += " --disable"
+
+        return ret
+
+    def eval_rules(self, ksdata, storage, report_only=False):
+        """:see: RuleHandler.eval_rules"""
+
+        messages = []
+
+        if self._kdump_enabled == None:
+            return []
+
+        if self._kdump_enabled == False:
+            msg = _("Kdump will be disabled on startup")
+        if self._kdump_enabled == True:
+            msg = _("Kdump will be enabled on startup")
+
+        messages.append(RuleMessage(self.__class__,
+                                    common.MESSAGE_TYPE_INFO, msg))
+
+        try:
+            if not report_only:
+                ksdata.addons.com_redhat_kdump.enabled = self._kdump_enabled
+        except AttributeError:
+            log.warning("com_redhat_kdump is not installed. "
+                        "Skipping kdump configuration")
+
+        return messages
+
+    def revert_changes(self, ksdata, storage):
+        """:see: RuleHander.revert_changes"""
+
+        try:
+            if ksdata.addons.com_redhat_kdump.enabled == False or \
+               self._kdump_enabled == False:
+                ksdata.addons.com_redhat_kdump.enabled = True
+        except AttributeError:
+            log.warning("com_redhat_kdump is not installed. "
+                        "Skipping reverting kdump configuration")
+
+        self._kdump_enabled = None
```

Comment:
_kdump_enabled is basically a tribool, right? unknown, true or false. this looks like it will only revert kdump disable but never kdump enable

Reply:
@mpreisler kinda by design as kdump is enabled by default.

Reply:
we shouldn't rely on that in the code logic IMO, it's pretty easy to do this in a fully generic way that will work even if Fedora or RHEL decide to disable kdump by default

Reply:
@mpreisler recommendations?

Reply:
I think the logic can be simpler:
If `self._kdump_enabled is not None`, just assign `ksdata.addons.com_redhat_kdump.enabled` the value of `self._kdump_default_enabled`

Reply:
Thanks @yuumasato! Good catch. Made the changes.